dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/loop/template/http/parser/HttpTaskDefinitionParser.java

```diff
@@ -60,7 +60,7 @@
     }
 
     protected @NonNull LoopTaskYamlDefinition parseYamlConfigFile(@NonNull String yamlConfigFile) throws IOException {
-        Yaml yaml = new Yaml(new Constructor(LoopTaskYamlDefinition.class));
+        Yaml yaml = new Yaml(new SafeConstructor());
```

Comment:
Can we fix this? @EricGao888

Reply:
In `package org.yaml.snakeyaml.constructor;`, we could see `public class Constructor extends SafeConstructor`. Therefore I think there is no need to change it.

Reply:
I'm not quite sure about it....

Reply:
> I'm not quite sure about it....

I just email an4er and request PR review from him / her to see if there is some extra stuff we need to handle this one.

Reply:
We can use `new SafeConstructor(new LoaderOptions())` to avoid this warning.

Reply:
First of all I think the code `Yaml yaml = new Yaml(new Constructor(LoopTaskYamlDefinition.class));` can't be changed to `Yaml yaml = new Yaml(new SafeConstructor());` . 
I would suggest to introduce the class [ClassFilterConstructor.java](https://github.com/apache/skywalking/blob/master/oap-server/server-library/library-util/src/main/java/org/apache/skywalking/oap/server/library/util/yaml/ClassFilterConstructor.java#L28), and then to replace the code in `AbstractK8sTaskExecutor` to be changed to
```
import java.util.List;

Yaml yaml = new Yaml(new ClassFilterConstructor(new Class[] {
        List.class
})).
```
This is for visibility, because here we expect the user's input to be of type list, you can also remove this map as it is already defined in the SafeConstructor
```
Yaml yaml = new Yaml(new ClassFilterConstructor(new Class[] {
}));
```
I didn't test the security of the parseYamlConfigFile function, so if you want to change it, you can also change to
```
Yaml yaml = new Yaml(new ClassFilterConstructor(new Class[] {
                LoopTaskYamlDefinition.class
        }));
```

Reply:
> Can we fix this? @EricGao888

Seems I misunderstood your point. Yes, we could fix this warning.

Reply:
> First of all I think the code `Yaml yaml = new Yaml(new Constructor(LoopTaskYamlDefinition.class));` can't be changed to `Yaml yaml = new Yaml(new SafeConstructor());` . I would suggest to introduce the class [ClassFilterConstructor.java](https://github.com/apache/skywalking/blob/master/oap-server/server-library/library-util/src/main/java/org/apache/skywalking/oap/server/library/util/yaml/ClassFilterConstructor.java#L28), and then to replace the code in `AbstractK8sTaskExecutor` to be changed to
> 
> ```
> import java.util.List;
> 
> Yaml yaml = new Yaml(new ClassFilterConstructor(new Class[] {
>         List.class
> })).
> ```
> 
> This is for visibility, because here we expect the user's input to be of type list, you can also remove this map as it is already defined in the SafeConstructor
> 
> ```
> Yaml yaml = new Yaml(new ClassFilterConstructor(new Class[] {
> }));
> ```
> 
> I didn't test the security of the parseYamlConfigFile function, so if you want to change it, you can also change to
> 
> ```
> Yaml yaml = new Yaml(new ClassFilterConstructor(new Class[] {
>                 LoopTaskYamlDefinition.class
>         }));
> ```

@an5er Thanks for your suggestions. It helps a lot.